sdk/src/main/scala/io/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -1068,15 +1070,41 @@ class EthService(
           if (start > end) {
             throw new RpcException(RpcError.fromCode(RpcCode.InvalidParams, "invalid block range"))
           }
+
+          val timeout = networkParams.getLogsQueryTimeout
+          var resultCount = 0
           // get the logs from all blocks in the range into one flat list
-          (start to end).flatMap(blockNumber => {
-            nodeView.history
-              .blockIdByHeight(blockNumber)
-              .map(ModifierId(_))
-              .flatMap(nodeView.history.getStorageBlockById)
-              .map(RpcFilter.getBlockLogs(stateView, _, query))
-              .get
-          })
+          val result: Try[Seq[EthereumLogView]] = Try {
+            val future = Future {
+              (start to end).flatMap(blockNumber => {
+                val logs = nodeView.history
+                  .blockIdByHeight(blockNumber)
+                  .map(ModifierId(_))
+                  .flatMap(nodeView.history.getStorageBlockById)
+                  .map(RpcFilter.getBlockLogs(stateView, _, query))
+                  .get
+
+                resultCount += logs.length
+                if (resultCount > networkParams.getLogsSizeLimit)
+                  throw new RpcException(RpcError.fromCode(RpcCode.InvalidParams, "Log response size exceeded. You can make eth_getLogs requests with up to a " + networkParams.getLogsSizeLimit +  " response size. Limit some parameters and try again."))
```

Comment:
Are these errors taken from other EVM chains?

Reply:
Calling the `eth_getLogs` rpc method for an extremely large number of blocks using Infura results in the following message:
```
{
    "jsonrpc": "2.0",
    "id": 1,
    "error": {
        "code": -32005,
        "data": {
            "from": "0x10114DC",
            "limit": 10000,
            "to": "0x10114F8"
        },
        "message": "query returned more than 10000 results. Try with this block range [0x10114DC, 0x10114F8]."
    }
}
```
BNB chain is much less verbose:
```
{
    "jsonrpc": "2.0",
    "id": 1,
    "error": {
        "code": -32005,
        "message": "limit exceeded"
    }
}
```
The code used (`32005`) seems to be a non-standard rpc code and it is reported in the [EIP-1474](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1474.md) but not directly present on _geth_. Let's discuss if necessary how to proceed in this case, personally I will keep the `32602` Invalid params error code.

Reply:
Current error messages are taken partly from Alchemy [[source](https://docs.alchemy.com/reference/eth-getlogs)]